NuRadioMC/simulation/simulation.py

```diff
@@ -401,752 +1339,325 @@ def __init__(self, inputfilename,
+                        for evt in output_buffer[sid].values():
+                            trace_start_time = evt.get_station().get_channel(channel_ids[0]).get_trace_start_time()
+                            # determine the trigger that was used to determine the readout window
+                            trigger_name = None
+                            trigger_time = None
+                            pre_trigger_time = None
+                            for trigger in evt.get_station().get_triggers().values():
+                                if trigger.get_pre_trigger_times() is not None:
+                                    pre_trigger_time = trigger.get_pre_trigger_times()
+                                    trigger_time = trigger.get_trigger_time()
+                                    trigger_name = trigger.get_name()
+                                    break
+
+                            assert(trigger_time is not None)
+                            assert(trigger_name is not None)
+                            assert(trace_start_time == trigger_time)
+
+                            for sim_channel in sim_station.get_channels_by_channel_id(channel_id):
+                                tt = sim_channel.get_times()
+                                t0 = tt[0]
+                                t1 = tt[-1]
+                                pre_trigger_time = triggerTimeAdjuster.get_pre_trigger_times(sim_channel.get_id())
+                                t0_readout = trigger_time - pre_trigger_time
+                                t1_readout = t0_readout + self._det.get_number_of_samples(sid, sim_channel.get_id()) * self._det.get_sampling_frequency(sid, sim_channel.get_id())
+
+                                # determine if the two intervals have any overlap
+                                if max(t0, t0_readout) < min(t1, t1_readout):
+                                    tmp_channel = NuRadioReco.framework.channel.Channel(sim_channel.get_id())
+                                    tmp_channel.set_trace(sim_channel.get_trace(), sim_channel.get_sampling_rate())
+                                    tmp_channel.set_trace_start_time(sim_channel.get_trace_start_time())
+                                    if(station.has_channel(sim_channel.get_id())):
+                                        channel = station.get_channel(sim_channel.get_id())
+                                        # logger.status(f"channel type {type(channel)}, sim_channel type {type(sim_channel)}")
+                                        combined_trace = channel + tmp_channel
+                                        channel.set_trace(combined_trace.get_trace(), combined_trace.get_sampling_rate())
+                                        channel.set_trace_start_time(combined_trace.get_trace_start_time())
+
+                                        # logger.status(f"channel type {type(channel)}")
```

Comment:
we could extent the "add_channel" interface or add a "combine_channel" interface.